bigtable/client/internal/wdeprecated_as_warnings.h

```diff
@@ -0,0 +1,28 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This file is special, it should not have include guards or #pragma once
+// directives
+
+#if defined(__clang__)
+#pragma clang diagnostic warning "-Wdeprecated-declarations"
```

Comment:
I understand the file name here is named after the flag, but `wdeprecated_as_warnings.h` still reads like a typo to me. Can we drop the leading `w`? Or call it something like `warnings_for_deprecations.h` or similar?

Reply:
`diagnostics_deprecated_as_warnings` ?  That creates a sensible pattern for other diagnostics.